doc/tutorials/changetracking.md

```diff
@@ -0,0 +1,112 @@
+
+## Basics
+
+The two headers you have to use are `kdbchangetracking.h` and `kdbdiff.h`.
+Those declare the `elektraChangeTracking*` and `elektraDiff*` methods.
+
+The two main data structures you will encounter are `ChangeTrackingContext` and `ElektraDiff`.
+
+## Getting the difference between KeySets
+
+If all you want to do is get the difference between two in-memory `KeySet` objects, use the function `elektraDiffCalculate`.
+
+```c
+KeySet * originalKeySet;
+KeySet * modifiedKeySet;
+Key * parentKey;
+
+// Calculate the difference of the keys below and including parentKey
+ElektraDiff * diff = elektraDiffCalculate (modifiedKeySet, originalKeySet, parentKey);
+
+// Extract useful information, see section 'Working with the diff'
+
+elektraDiffDel (diff);
+```
+
+## Getting the changes within a transaction in a plugin
+
+If you are writing a plugin, you can use `elektraChangeTrackingGetContextFromPlugin` to get the current `ChangeTrackingContext`.
+Then, use `elektraChangeTrackingCalculateDiff` to calculate the changes to the KDB.
+
+```c
+int myPluginSet (Plugin * handle, KeySet * returned, Key * parentKey)
+{
+	ChangeTrackingContext * ctx = elektraChangeTrackingGetContextFromPlugin (handle);
+	ElektraDiff * diff = elektraChangeTrackingCalculateDiff (returned, ctx, parentKey);
+
+	// Extract useful information, see section 'Working with the diff'
+
+	elektraDiffDel (diff);
+	return 1;
+}
+```
+
+## Getting the changes as an application developer
+
+If you are elektrifying your app, you can use `elektraChangeTrackingGetContextFromKdb` to get the current `ChangeTrackingContext` for your KDB instance.
+Then, use `elektraChangeTrackingCalculateDiff` to calculate the changes to the KDB.
+
+```c
+KDB * kdb;
+
+ChangeTrackingContext * ctx = elektraChangeTrackingGetContextFromPlugin (kdb);
+ElektraDiff * diff = elektraChangeTrackingCalculateDiff (returned, ctx, parentKey);
+
+// Extract useful information, see section 'Working with the diff'
+
+elektraDiffDel (diff);
+```
+
+## Working with the diff
+
+Congratulations! You've got your diff!
+Now what?
+Elektra provides you with different functions to determine added, removed and modified keys, as well as added, removed and modified metadata.
+
+You can use the following methods to reason about changes to the keyset as a whole:
+
+```c
+// Get keys that are in modifiedKeySet but not in originalKeySet
+KeySet * addedKeys = elektraDiffGetAddedKeys (diff);
+
+// Get keys that are in originalKeySet but not in modifiedKeySet
+KeySet * removedKeys = elektraDiffGetRemovedKeys (diff);
+
+// Get keys that are in both KeySets, but have either different values or different metadata.
+// The returned keys will have the old values and metadata
+KeySet * modifiedKeys = elektraDiffGetModifiedKeys (diff);
+
+ksDel (addedKeys);
+ksDel (removedKeys);
+ksDel (modifiedKeys);
+```
+
+Use these methods to get information about single keys:
+
+```c
+Key * myKey;
```

Comment:
It is unclear where myKey comes from or what it might contain.

Reply:
I added a comment to clarify now. It is the key that we want to check. So for example, if the key has been modified, or what the old metadata of the key looked like.